gather-system-info.sh

```diff
@@ -20,6 +20,13 @@ fi
 
 gnome-shell --version
 
+echo -n "Display server: "
+if [ "${XDG_SESSION_TYPE}" = "wayland" ]; then
```

Comment:
Just want to share one of my preferences for shell scripting, not asking for a change:
```suggestion
if test "${XDG_SESSION_TYPE}" = "wayland"; then
```
I really like the educational benefit of using `test` rather than single brackets.

- Helps build an intuition for "`if` just switches on the status code of the command" so it doesn't become a separate concept for people to understand when you want to use a non-builtin there.
- Possible to `man test` vs having to do a web search for how brackets work in [ba]sh
- Things like `-gt` being flags actually makes sense.
- Unsure what the benefit of using brackets is when they only save a single character

Reply:
Actually it is better to use `[[` instead of `[`. Because `[` is a synonym for `test` and therefore a command. And it is therefore also evaluated like a command, which has in some cases subtle, inconsistent or surprising behavior. `[[` is a keyword and therefore everything in `[[ ... ]]` is treated as "one thing" and evaluated in a more consistent way. Usually there is no difference at all. But in cases there is, it is usually not at all obvious what is going on.

Also using `[[` you can also use `<`, `>`, `=~` (for regex), `==` and `!=`. So in that sense it behaves more like other languages.

For most things bash I usually just refer to https://devhints.io/bash Since I can't remember everything and I think the syntax is not very intuitive. But you can also use `man bash` and search for `CONDITIONAL EXPRESSIONS`.